lib/publiccloud/ssh_interactive.pm

```diff
@@ -67,4 +67,12 @@ sub ssh_interactive_leave {
     $testapi::distri->set_standard_prompt('root');
 }
 
+sub ssh_interactive_post_fail {
+    select_console 'tunnel-console', await_console => 0;
+    send_key "ctrl-c";
+    send_key "ret";
+    assert_script_run('cd /root/terraform');
+    script_run('terraform destroy -no-color -auto-approve', 240);
```

Comment:
What do you think. Is it worth to take the work and store the args in that object and call the provider cleanup like we do here https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/publiccloud/ssh_interactive_end.pm#L29 .

Out of the blue, I would like that effort, for that hypothetical scenario, where we change from terraform to some other public-cloud instantiation... see ticket https://progress.opensuse.org/issues/63604

Maybe the easiest way would be to create a class which extends the console and you only change the `base` statement in all that files, wdyt?

Reply:
> What do you think. Is it worth to take the work and store the args in that object and call the provider cleanup like we do here https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/publiccloud/ssh_interactive_end.pm#L29 

I do not suggest that unless we solve the issue of preserving the `$args` within the YAML scheduler.

> Maybe the easiest way would be to create a class which extends the console and you only change the base statement in all that files, wdyt?

It should be possible, I will do it.